pkgs/build-support/rust/default.nix

```diff
@@ -173,36 +173,42 @@ stdenv.mkDerivation (filteredArgs // {
         --frozen ${concatStringsSep " " cargoBuildFlags}
     )
 
-    # rename the output dir to a architecture independent one
-    mapfile -t targets < <(find "$NIX_BUILD_TOP" -type d | grep '${releaseDir}$')
-    for target in "''${targets[@]}"; do
-      rm -rf "$target/../../${buildType}"
-      ln -srf "$target" "$target/../../"
-    done
-
     runHook postBuild
+
+    # This needs to be done after postBuild: packages like `cargo` do a pushd/popd in
+    # the pre/postBuild-hooks that need to be taken into account before gathering
+    # all binaries to install.
+    bins=$(find $releaseDir \
```

Comment:
I see a few issues here: if someone has their own buildPhase, this variable will be undefined (we could check though if it's declared and if not, rerun the `find ...` command). Any further ideas?

Reply:
Would it be a bad idea to append this to `postBuild` itself? Something like:

```nix
postBuild = args.postBuild or "" + ''
    bins=$(find $releaseDir \
      -maxdepth 1 \
      -type f \
      -executable ! \( -regex ".*\.\(so.[0-9.]+\|so\|a\|dylib\)" \))
'';
```

This way, if a package defines a `postBuild`, it executes that `postBuild` and then also executes the `find`.

Reply:
Sorry, forgot to reply here today! The idea sounds reasonable to me, will test it locally and push it if it's fine :)

Reply:
Should be fixed now (unless I misunderstood your suggestion :)).

Reply:
I reverted this change again. This would break packages like `uutils-coreutils` with custom `build`/`installPhase` where you'd have to set `postBuild = ""` otherwise.

Reply:
Unfortunate. What if you did the above suggestion but for `buildPhase` instead of `postBuild`? Like

```nix
buildPhase = with builtins; args.buildPhase or ''
      # the stuff that's already there
    '' + ''
      bins=$(find $releaseDir \
        -maxdepth 1 \
        -type f \
        -executable ! \( -regex ".*\.\(so.[0-9.]+\|so\|a\|dylib\)" \))
    '';
```

Do you think that would work for `uutils-coreutils` (and the others having problems)?